File: backend/src/main/java/backend/config/PfConfigService.java
package backend.config;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import backend.repository.PfConfigRepository;
import backend.repository.entity.PfConfig;

@Component
public class PfConfigService {

	@Autowired private PfConfigRepository repository;

	public PfConfig get() {
		Optional<PfConfig> result = repository.findById(new PfConfig().getId());
		return result.isPresent() ? result.get() : new PfConfig();
	}

	public PfConfig put(PfConfig entity) {
		return repository.save(entity);
	}
}

File: backend/src/main/kotlin/backend/config/PfConfigService.kt
package backend.config

import backend.repository.PfConfigRepository
import backend.repository.entity.PfConfig
import org.springframework.stereotype.Component

@Component
class PfConfigService (private val repository: PfConfigRepository) {

    fun get(): PfConfig {
        val result = repository.findById(PfConfig().id)
        return if (result.isPresent) result.get() else PfConfig()
    }

    fun put(entity: PfConfig): PfConfig {
        return repository.save(entity)
    }
}